### DISCUSSION TITLE
NOT with parentheses for complex expression?

### DISCUSSION THREAD

**Discussion by 6xzo** [SeaQL/sea-query#143] [Q&A] [answered]
Passing a complex `Expr` to `Expr::not`, I expected parentheses in the output, to respect the precedence ordering.

```
use sea_query::*;

fn main() {
    let t1 = Expr::col(Alias::new("a")).eq(1);
    let t2 = Expr::col(Alias::new("b")).eq(1);
    let expr = t1.and(t2);
    let expr = Expr::expr(expr).not();
    let query = Query::select()
        .and_where(expr)
        .to_string(MysqlQueryBuilder);
    println!("{}", query);
}
```

prints:

```
SELECT WHERE NOT (`a` = 1) AND (`b` = 1)
```

I expected:

```
SELECT WHERE NOT ( (`a` = 1) AND (`b` = 1) )
```

Is there a way to achieve what I expected?

I tried to use Conditions, but cannot figure out how to apply `not` to them.

**Comment by tyt2y3** [comment, marked as answer]
Hi there. You are right. `Condition` is the right way for building complex logic trees.
This feature is on our roadmap https://github.com/SeaQL/sea-orm/issues/147
We hope to implement it soon, or if *you* (whom are reading) are interested in implementing,
the historic reference of `Condition` is #47

**Reply by 6xzo** [reply]
Thank you
